database/migration_runner_env.go

```diff
@@ -0,0 +1,87 @@
+package database
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+)
+
+// Constants representing environment variable keys for migration configuration.
+const (
+	envKeyPrefix       = "MIGRATION"
+	envKeyDsn          = "DSN"
+	envKeyOp           = "OPERATION"
+	envKeyForceVersion = "FORCE_VERSION"
+	envKeyFilesDir     = "FILES_DIR"
+)
+
+func envKey(key string) string {
+	return fmt.Sprintf("%s_%s", envKeyPrefix, key)
+}
+
+func readForceVersion() (int, error) {
+	forceVersionRaw, ok := os.LookupEnv(envKey(envKeyForceVersion))
+	if !ok {
+		return 0, nil
+	}
+
+	forceVersion, err := strconv.Atoi(forceVersionRaw)
+	if err != nil {
+		return 0, fmt.Errorf("convert forceVersion: %v", err)
+	}
+
+	return forceVersion, nil
+}
+
+// RunMigrationsFromEnv reads migration configuration from environment variables,
+// creates a MigrationRunner, and runs the specified migration operation.
+func RunMigrationsFromEnv(logger logger) error {
+	dsn, ok := os.LookupEnv(envKey(envKeyDsn))
+	if !ok {
+		return fmt.Errorf("missing env: %s", envKey(envKeyDsn))
+	}
+
+	operation, ok := os.LookupEnv(envKey(envKeyOp))
+	if !ok {
+		return fmt.Errorf("missing env: %s", envKey(envKeyOp))
+	}
+
+	forceVersion, err := readForceVersion()
```

Comment:
[minor] this is required only for "force" operation, in other cases it's just ignored.

Reply:
yes and error will occur here only when related env var is set